libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2994,18 +2991,17 @@ string YulUtilFunctions::arrayConversionFunction(ArrayType const& _from, ArrayTy
 				"body",
 				Whiskers(R"(
 					// Copy the array to a free position in memory
+					converted :=
 					<?fromStorage>
-						converted := <arrayStorageToMem>(value)
+						<arrayStorageToMem>(value)
 					</fromStorage>
 					<?fromCalldata>
-						converted := <allocateMemoryArray>(length)
-						<copyToMemory>(value, add(converted, 0x20), length)
+						<abiDecode>(value, calldatasize())
```

Comment:
Should make use of the `length` parameter if provided.

Reply:
I don't understand exactly how to use it here, trying to find an example usage with similar type

Reply:
```suggestion
						<abiDecode>(value, add(value, length))
```
does not work?

Reply:
Haven't really thought this through...

Reply:
I was thinking it should work like that, but it doesn't